### PR TITLE
Add "[createdump]" prefix to all createdump output and logging for VS4Mac

### PR DESCRIPTION
**Pull request by mikem8361**
# Customer Impact

There are a few cases where no crash report or core dump is not generated and the crash report is missing native stack frames. The current logging isn't enough and VS4Mac has no way to distinguish createdump's output to include it in their reports. To help diagnose and improve dump collection for VS4Mac, the createdump's messages and logging needed a prefix: "[createdump]".

Add verbose createdump logging env var COMPlus_CreateDumpVerboseDiagnostics.  

# Testing

The VS4Mac team has verified this change.

# Risk

Low. Just createdump changes except one minor PAL change.

### CODE REVIEW

**Comment by mikem8361** [comment]
@jeffschwMSFT can we consider this for next service request and for the VS4Mac 6.0 branch/build?

@sgmunn, @kdubau can you verify this once I get a build for you?

File: src/coreclr/pal/src/thread/process.cpp [hunk 1]
```diff
@@ -3286,13 +3286,19 @@ PROCAbortInitialize()
         char* dumpType = getenv("COMPlus_DbgMiniDumpType");
         char* diagStr = getenv("COMPlus_CreateDumpDiagnostics");
         BOOL diag = diagStr != nullptr && strcmp(diagStr, "1") == 0;
+        char* verboseStr = getenv("COMPlus_CreateDumpVerboseDiagnostics");
```
Comment (jkoritzinsky) [inline src/coreclr/pal/src/thread/process.cpp:3289]: Is there any particular reason that we haven't aligned these environment variables with the rest of our config variables (supporting a `DOTNET_` prefix in addition to a `COMPlus_` prefix)?

Comment (hoyosjs) [reply src/coreclr/pal/src/thread/process.cpp:3289]: It's been fixed, didn't make it into 6.0 https://github.com/dotnet/runtime/blob/a71d6f1c99595a6ceafe42ed8bc30fdf031e9b7e/src/coreclr/pal/src/thread/process.cpp#L3309 Any config in the PAL is not going to accept DOTNET_ in 6.0. Granted, it's very few

Comment (jkoritzinsky) [reply src/coreclr/pal/src/thread/process.cpp:3289]: Sorry I didn't realize this was going into 6.0. Sorry for the confusion.

**Review by jeffschwMSFT — APPROVED** [review]
Approved.  Pending code review feedback.  We should take for consideration in 6.0.x

File: src/coreclr/debug/createdump/main.cpp [hunk 2]
```diff
@@ -152,21 +152,21 @@ int __cdecl main(const int argc, const char* argv[])
         {
             if (::GetTempPathA(MAX_LONGPATH, tmpPath) == 0)
             {
-                fprintf(stderr, "GetTempPath failed (0x%08x)", ::GetLastError());
+                printf_error("GetTempPath failed (0x%08x)", ::GetLastError());
```
Comment (hoyosjs) [inline src/coreclr/debug/createdump/main.cpp:155]: nit: should lines like this have newlines?

**Review by hoyosjs — APPROVED** [review]

**Comment by jeffschwMSFT** [comment]
@safern can we move this item over to the vs4mac release branch?

**Comment by safern** [comment]
cc: @carlossanlop

**Comment by carlossanlop** [comment]
Can you please elaborate, @jeffschwMSFT? Do you mean we need this change needs to target a different branch instead of release/6.0?

**Comment by jeffschwMSFT** [comment]
We are good to merge here.  The vs4mac release branch is no longer used.

**Comment by carlossanlop** [comment]
Ok perfect, thanks for the clarification.

The PR has been signed-off, the `servicing-approved` label applied, and the CI is green. Merging now.

**Pull request merged**